### PR TITLE
chore: fetch number of fraction for currency dynamically

### PR DESCRIPTION
**Pull request by nicolasburtey**

### CODE REVIEW

File: app/graphql/cache.ts [hunk 1]
```diff
@@ -61,6 +63,33 @@ const getWallets = ({
...
+  const resCurrency = cache.readQuery<DisplayCurrencyQuery>({
+    query: DisplayCurrencyDocument,
+  })
+  const displayCurrency = resCurrency?.me?.defaultAccount?.displayCurrency
+
+  if (!displayCurrency) {
+    return { displayCurrency: null, fractionDigits: null }
+  }
+
+  const res2 = cache.readQuery<CurrencyListQuery>({
```
Comment (UncleSamtoshi) [inline app/graphql/cache.ts:78]: `res2` should be renamed to something like `resCurrencyList`

File: app/graphql/cache.ts [hunk 2]
```diff
@@ -61,6 +63,33 @@ const getWallets = ({
...
+    query: CurrencyListDocument,
+    variables: { currency: displayCurrency },
+  })
+  const currencyList = res2?.currencyList ?? []
+  if (!res2?.currencyList) {
+    return { displayCurrency: null, fractionDigits: null }
+  }
+
+  const fractionDigits =
```
Comment (UncleSamtoshi) [inline app/graphql/cache.ts:87]: Does it make sense to have a default to 2  here, in other places if we can't find the value we return null. Something is wrong with the currency list if a value can't be found.

File: app/graphql/cache.ts [hunk 3]
```diff
@@ -61,6 +63,33 @@ const getWallets = ({
...
+
+  if (!displayCurrency) {
+    return { displayCurrency: null, fractionDigits: null }
+  }
+
+  const res2 = cache.readQuery<CurrencyListQuery>({
+    query: CurrencyListDocument,
+    variables: { currency: displayCurrency },
+  })
+  const currencyList = res2?.currencyList ?? []
```
Comment (UncleSamtoshi) [inline app/graphql/cache.ts:82]:
```suggestion
  const currencyList = res2?.currencyList
  if (!currencyList) ....
```

File: app/hooks/use-display-currency.ts [hunk 4]
```diff
@@ -53,6 +54,10 @@ export const useDisplayCurrency = () => {
     )
   }, [dataCurrencyList, displayCurrency])
 
+  const fractionDigits = useMemo(() => {
```
Comment (UncleSamtoshi) [inline app/hooks/use-display-currency.ts:57]: It takes longer to memoize this then it does to just have `const fractionDigits = displayCurrencyInfo.fractionDigits` which is a simple assignment vs a function call to useMemo every render where it has to check for equality of dependencies.

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]

File: app/graphql/cache.ts [hunk 5]
```diff
@@ -61,6 +63,35 @@ const getWallets = ({
...
+    variables: { currency: displayCurrency },
+  })
+  const currencyList = resCurrencyList?.currencyList
+
+  if (!currencyList) {
+    return { displayCurrency: null, fractionDigits: null }
+  }
+
+  const fractionDigits =
+    currencyList.find((currency) => currency.id === displayCurrency)?.fractionDigits ??
```
Comment (UncleSamtoshi) [inline app/graphql/cache.ts:89]: If fraction digits is 0 here it will be falsy and then the value will be null.

File: app/graphql/cache.ts [hunk 6]
```diff
@@ -171,20 +202,16 @@ export const createCache = () =>
...
-              // could we have a way to not fetch DisplayCurrencyQuery?
-              const resCurrency = cache.readQuery<DisplayCurrencyQuery>({
-                query: DisplayCurrencyDocument,
-              })
-              const currency = resCurrency?.me?.defaultAccount?.displayCurrency
+              const { displayCurrency, fractionDigits } =
+                getDisplayCurrencyAndFractionDigits(cache)
 
-              if (!currency) {
+              if (!displayCurrency || !fractionDigits) {
```
Comment (UncleSamtoshi) [inline app/graphql/cache.ts:208]: FractionDigits could be 0 here so the check should `typeof fractionDigits !== "number"`

File: app/graphql/cache.ts [hunk 7]
```diff
@@ -210,20 +235,16 @@ export const createCache = () =>
...
-              // could we have a way to not fetch DisplayCurrencyQuery?
-              const resCurrency = cache.readQuery<DisplayCurrencyQuery>({
-                query: DisplayCurrencyDocument,
-              })
-              const currency = resCurrency?.me?.defaultAccount.displayCurrency
+              const { displayCurrency, fractionDigits } =
+                getDisplayCurrencyAndFractionDigits(cache)
 
-              if (!currency) {
+              if (!displayCurrency || !fractionDigits) {
```
Comment (UncleSamtoshi) [inline app/graphql/cache.ts:241]: same here

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]

File: app/graphql/cache.ts (hunk 5 above)
Comment (UncleSamtoshi) [reply app/graphql/cache.ts:89]: You can just remove the `?? null`

Comment (nicolasburtey) [reply app/graphql/cache.ts:89]: not with ??:

```
% node
Welcome to Node.js v18.14.0.
Type ".help" for more information.
> 0 ?? "a"
0
> 0 || "a"
'a'
```

Comment (UncleSamtoshi) [reply app/graphql/cache.ts:89]: I thought it behaved the same as `||`. My mistake.

Comment (nicolasburtey) [reply app/graphql/cache.ts:89]: if I remove `?? null` then the return type of undefined. then I have to check both for `null` and `undefined` in the calling function, which I don't want to do

**Pull request merged**